ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -508,6 +520,181 @@ const triggerSet: TriggerSet<Data> = {
+          ids: [ids],
+        })).combatants;
+      },
+      run: (data, matches) => {
+        const x = data.combatantData[0]?.PosX;
+        if (x === undefined)
+          return;
+        const y = data.combatantData[0]?.PosY;
+        if (y === undefined)
+          return;
+
+        if (x === 75 && y === 97) {
+          if (matches.id === '82F1')
+            data.engravement1BeamsPosMap.set('NE', 'dark');
+          else
+            data.engravement1BeamsPosMap.set('NE', 'light');
+        } else if (x === 97 && y === 75) {
+          if (matches.id === '82F1')
+            data.engravement1BeamsPosMap.set('SW', 'dark');
+          else
+            data.engravement1BeamsPosMap.set('SW', 'light');
+        } else if (x === 103 && y === 75) {
+          if (matches.id === '82F1')
+            data.engravement1BeamsPosMap.set('SE', 'dark');
+          else
+            data.engravement1BeamsPosMap.set('SE', 'light');
+        } else if (x === 125 && y === 97) {
+          if (matches.id === '82F1')
+            data.engravement1BeamsPosMap.set('NW', 'dark');
+          else
+            data.engravement1BeamsPosMap.set('NW', 'light');
+        } else if (x === 125 && y === 103) {
+          if (matches.id === '82F1')
+            data.engravement1BeamsPosMap.set('SW', 'dark');
+          else
+            data.engravement1BeamsPosMap.set('SW', 'light');
+        } else if (x === 103 && y === 125) {
+          if (matches.id === '82F1')
+            data.engravement1BeamsPosMap.set('NE', 'dark');
+          else
+            data.engravement1BeamsPosMap.set('NE', 'light');
+        } else if (x === 97 && y === 125) {
+          if (matches.id === '82F1')
+            data.engravement1BeamsPosMap.set('NW', 'dark');
+          else
+            data.engravement1BeamsPosMap.set('NW', 'light');
+        } else if (x === 75 && y === 103) {
+          if (matches.id === '82F1')
+            data.engravement1BeamsPosMap.set('SE', 'dark');
+          else
+            data.engravement1BeamsPosMap.set('SE', 'light');
+        }
+      },
+    },
+    {
+      id: 'P12S Engravement 1 Beam',
+      type: 'StartsUsing',
+      netRegex: { id: ['82F[12]'], source: 'Anthropos' },
+      condition: (data) => data.engravementCounter === 1,
+      alertText: (data, matches, output) => {
+        if (data.me === matches.target) {
+          if (matches.id === '82F1')
+            return output.lightBeam!();
+          return output.darkBeam!();
+        }
+      },
+      outputStrings: {
+        lightBeam: {
+          en: 'light beam',
+          cn: '引导光激光',
+        },
+        darkBeam: {
+          en: 'dark beam',
+          cn: '引导暗激光',
+        },
+      },
+    },
+    {
+      id: 'P12S Engravement 1 Tower Drop Notifier',
+      type: 'GainsEffect',
+      netRegex: { effectId: ['DFB', 'DFC'] },
+      condition: (data) => data.engravementCounter === 1,
+      delaySeconds: 1, // Make sure Drop Tower Pos Tracker is done
```

Comment:
How close are these in time? Would it make sense to have a collector for `DFB`/`DFC` and do the callout in `P12S Engravement 1 Drop Tower Pos Tracker` as soon as the locations are known?

Reply:
In combat, the buff starts at 1:06s, and then soak tower at 1:16s, with 10s of time to run to position, which is delayed for 1s to allow the tracker above to gather tower information. Where the delay takes into account the network delay so set the display after 1s, I think 9s for run is enough.

![image](https://github.com/quisquous/cactbot/assets/12003087/de7269c0-6ad3-48e0-a78a-2fe9107a7381)
![image](https://github.com/quisquous/cactbot/assets/12003087/4dfb1241-8680-424a-b7b7-6674e57876a5)

Reply:
I actually tried it, and it seemed like a better change

https://github.com/quisquous/cactbot/blob/edc4fe0647ad6fb893533fe4b8468956aadd49af/ui/raidboss/data/06-ew/raid/p12s.ts#L747-L748